core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionResponse.kt

```diff
@@ -9,22 +9,47 @@ import fr.sncf.osrd.utils.json.UnitAdapterFactory
 import java.time.ZonedDateTime
 
 class ConflictDetectionResponse(
+    /**
+     * List of all detected conflicts, if any. Conflicts are grouped between each set of conflicting
+     * train/work schedule IDs. (i.e. there's only one conflict entry between trains (A, B), but
+     * there may be a different one for trains (A, B), or (B, C, D)).
+     */
     val conflicts: Collection<Conflict>,
 )
 
+/**
+ * One conflict between a set of trains. If a given set of train is conflicting over a continuous
+ * time range, only one conflict is returned (with a longer requirement list).
+ */
 class Conflict(
+    /** List of train IDs for this given conflict. Can't be empty. */
     @Json(name = "train_ids") val trainIds: Collection<Long>,
+    /** List of work schedule IDs for this given conflict, if any. */
     @Json(name = "work_schedule_ids") val workScheduleIds: Collection<Long>,
+    /** Start of the conflict time range. This is the *union* of all the conflicting time ranges. */
     @Json(name = "start_time") val startTime: ZonedDateTime,
+    /** End of the conflict time range. See `start_time`. */
     @Json(name = "end_time") val endTime: ZonedDateTime,
+    /** One of "SPACING" or "ROUTING" depending on the kind of conflicting resource. */
     @Json(name = "conflict_type")
     val conflictType: ConflictDetectionEndpoint.ConflictDetectionResult.Conflict.ConflictType,
+    /** List of all conflicting requirements. Can't be empty. */
     @Json(name = "requirements") val requirements: Collection<ConflictRequirement>,
 )
 
+/**
+ * One conflicting requirement. The zone is used by at least two of the conflicting
+ * [trains / work schedule] in the given time range. This time range is the *union* and *not the
+ * intersection* of the time during which they are used by at least one train. i.e. if a train uses
```

Comment:
I do believe here, it's either a zone or a resource which is used.
```suggestion
 * intersection* of the time during which it is used by at least one train, i.e., if a train uses
```

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/9391/commits/649f0b097ad6f6412d4e5a8a1842faef22ee3e70